src/math_util_2.c

```diff
@@ -478,117 +479,124 @@ UNUSED void func_80041A70(void) {
     arg0[1][1] = 1.0f;
     arg0[2][2] = 1.0f;
     arg0[1][0] = 0.0f;
     arg0[2][0] = 0.0f;
     arg0[0][1] = 0.0f;
-    arg0[3][0] = arg1;
+    arg0[3][0] = x;
     arg0[2][1] = 0.0f;
     arg0[0][2] = 0.0f;
     arg0[1][2] = 0.0f;
     arg0[3][2] = 0.0f;
-    arg0[3][1] = arg2;
+    arg0[3][1] = y;
     arg0[0][3] = 0.0f;
     arg0[1][3] = 0.0f;
     arg0[2][3] = 0.0f;
     arg0[3][3] = 1.0f;
-}
 
-void func_80041AD8(Mat4 arg0, u16 arg1) {
-    f32 sp1C = sins(arg1);
-    f32 temp_f0 = coss(arg1);
-    
-    arg0[0][0] = temp_f0;
-    arg0[1][0] = -sp1C;
-    arg0[1][1] = temp_f0;
-    arg0[0][1] = sp1C;
-    arg0[2][0] = 0.0f;
-    arg0[3][0] = 0.0f;
-    arg0[2][1] = 0.0f;
-    arg0[3][1] = 0.0f;
-    arg0[0][2] = 0.0f;
-    arg0[1][2] = 0.0f;
-    arg0[3][2] = 0.0f;
-    arg0[0][3] = 0.0f;
-    arg0[1][3] = 0.0f;
-    arg0[2][3] = 0.0f;
-    arg0[2][2] = 1.0f;
-    arg0[3][3] = 1.0f;
-}
-
-void func_80041B68(Mat4 arg0, f32 arg1) {
-    arg0[1][0] = 0.0f;
-    arg0[2][0] = 0.0f;
-    arg0[3][0] = 0.0f;
-    arg0[0][1] = 0.0f;
-    arg0[2][1] = 0.0f;
-    arg0[3][1] = 0.0f;
-    arg0[0][2] = 0.0f;
-    arg0[1][2] = 0.0f;
-    arg0[3][2] = 0.0f;
-    arg0[0][3] = 0.0f;
-    arg0[1][3] = 0.0f;
-    arg0[2][3] = 0.0f;
-    arg0[2][2] = 1.0f;
-    arg0[3][3] = 1.0f;
-    arg0[0][0] = arg1;
-    arg0[1][1] = arg1;
-}
-
-void func_80041BBC(Mat4 arg0, u16 arg1, f32 arg2) {
-    f32 sp1C = sins(arg1);
-    f32 temp_f12 = coss(arg1) * arg2;
-
-    arg0[2][0] = 0.0f;
-    arg0[0][0] = temp_f12;
-    arg0[1][1] = temp_f12;
-    arg0[2][1] = 0.0f;
-    arg0[1][0] = -sp1C * arg2;
-    arg0[0][2] = 0.0f;
-    arg0[1][2] = 0.0f;
-    arg0[0][1] = sp1C * arg2;
-    arg0[3][2] = 0.0f;
-    arg0[0][3] = 0.0f;
-    arg0[1][3] = 0.0f;
-    arg0[2][3] = 0.0f;
-    arg0[3][0] = 1.0f;
-    arg0[3][1] = 1.0f;
-    arg0[2][2] = 1.0f;
-    arg0[3][3] = 1.0f;
+    /*
+     * 1 0 0 x
+     * 0 1 0 y
+     * 0 0 1 0
+     * 0 0 0 1    
+     */
+}
+
+void mtxf_u16_rotate_z(Mat4 dest, u16 angle) {
+    f32 sin_theta = sins(angle);
+    f32 cos_theta = coss(angle);
+
+    dest[0][0] = cos_theta;
+    dest[1][0] = -sin_theta;
+    dest[1][1] = cos_theta;
+    dest[0][1] = sin_theta;
+    dest[2][0] = 0.0f;
+    dest[3][0] = 0.0f;
+    dest[2][1] = 0.0f;
+    dest[3][1] = 0.0f;
+    dest[0][2] = 0.0f;
+    dest[1][2] = 0.0f;
+    dest[3][2] = 0.0f;
+    dest[0][3] = 0.0f;
+    dest[1][3] = 0.0f;
+    dest[2][3] = 0.0f;
+    dest[2][2] = 1.0f;
+    dest[3][3] = 1.0f;
+}
+
+void mtxf_scale_x_y(Mat4 dest, f32 scale) {
+    dest[1][0] = 0.0f;
+    dest[2][0] = 0.0f;
+    dest[3][0] = 0.0f;
+    dest[0][1] = 0.0f;
+    dest[2][1] = 0.0f;
+    dest[3][1] = 0.0f;
+    dest[0][2] = 0.0f;
+    dest[1][2] = 0.0f;
+    dest[3][2] = 0.0f;
+    dest[0][3] = 0.0f;
+    dest[1][3] = 0.0f;
+    dest[2][3] = 0.0f;
+    dest[2][2] = 1.0f;
+    dest[3][3] = 1.0f;
+    dest[0][0] = scale;
+    dest[1][1] = scale;
+}
+
+void mtxf_rotate_z_scale_x_y(Mat4 dest, u16 angle, f32 scale) {
```

Comment:
Could you please add `UNUSED` to this func?

`UNUSED void mtxf...`

(yea, unfortunately it's called by an unused func)